datahub-graphql-core/src/main/java/com/linkedin/datahub/graphql/GmsGraphQLEngine.java

```diff
@@ -229,6 +249,11 @@ private static void configureDatasetResolvers(final RuntimeWiring.Builder builde
                                     DOWNSTREAM_LINEAGE_TYPE,
                                     (env) -> ((Dataset) env.getSource()).getUrn()))
                     )
+                    .dataFetcher("genericDownstreamLineage", new AuthenticatedResolver<>(
+                            new LoadableTypeResolver<>(
+                                    GENERIC_DOWNSTREAM_LINEAGE_TYPE,
+                                    (env) -> ((Dataset) env.getSource()).getUrn()))
```

Comment:
Are we always attempting to fetch a downstream Dataset? 

If not, we may want to think more about this case...

Reply:
this is saying we are always attempting to fetch generic downstream lineage of a dataset, not that the results will always be datasets. See the wiring above- this is specifically wired for Dataset.genericDownstreamLineage

Reply:
Got it!